### PR TITLE
X509_STORE_new: memory needs to free

### PR DESCRIPTION
**Pull request by PeiweiHu**

### CODE REVIEW

File: demos/cms/cms_ver.c [hunk 1]
```diff
@@ -27,6 +27,8 @@ int main(int argc, char **argv)
     /* Set up trusted CA certificate store */
 
     st = X509_STORE_new();
+    if (!st)
```
Comment (t8m) [inline demos/cms/cms_ver.c:30]: Please use `st == NULL` instead.  You can unify the style in the below ifs as well, if you want.

File: demos/smime/smver.c [hunk 2]
```diff
@@ -27,6 +27,8 @@ int main(int argc, char **argv)
     /* Set up trusted CA certificate store */
 
     st = X509_STORE_new();
+    if (!st)
```
Comment (t8m) [inline demos/smime/smver.c:30]: Same thing applies here.

**Review by t8m — CHANGES_REQUESTED** [review]

File: demos/cms/cms_ver.c (hunk 1 above)
Comment (PeiweiHu) [reply demos/cms/cms_ver.c]: done, Thanks for you review.

**Review by paulidale — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Merged to master and 3.0 branches. Thank you.

**Pull request closed without merging**